pydatastructs/graphs/algorithms.py

```diff
@@ -548,3 +550,68 @@ def strongly_connected_components(graph, algorithm):
         "isn't implemented for finding strongly connected components."
         %(algorithm, graph._impl))
     return getattr(algorithms, func)(graph)
+
+
+def random_walk(graph, start, path_length, alpha = None):
+    """
+    Executes random walk from the starting node and given graph.
+
+    Parameters
+    ==========
+
+    graph: Graph
+        The graph on which random walk needs to implemented.
+    start: Starting GraphNode
+        The node/vertex from where random walk will begin.
+    path_length:
+        The path length of random walk.
+    alpha:
```

Comment:
How does this parameter makes random walk more random? Any references?

Reply:
Here `alpha` is a floating point number between 0 and 1. The `random.random()` function generates random floating number between 0 and 1. For example, if value if `alpha` is as low as 0.001, the chances are more that `random.random()` will generate a floating point number greater than `alpha`. As a result of this, the `RandomWalk` will not restart. On the other side, if `alpha` has value close to 1, the `RandomWalk`, will restart more frequently due to which it can go much away from the `start` node, reducing randomness.
This method to control randomness was used in random walk [here](https://github.com/phanein/deepwalk).

Reply:
Why would a random walk need a restart?

Reply:
It is optional. If it has no use, I will remove this part.